src/azure-cli/az.completion.ps1

```diff
@@ -0,0 +1,67 @@
+$AzCommand = "az"
+
+$AzArgCompleteScriptBlock = {
+    param($wordToComplete, $commandAst, $cursorPosition)
+
+    # In case of scripts, this object hold the current line after string conversion
+    $line = "$commandAst"
+
+    # The behaviour of completion should depend on the trailing spaces in the current line:
+    # * "command subcommand " --> TAB --> Completion items parameters/sub-subcommands of "subcommand"
+    # * "command subcom" --> TAB --> Completion items to extend "subcom" into matching subcommands.
+    # $line never contains the trailing spaces. However, $cursorPosition is the length of the original
+    # line (with trailing spaces) in this case. This comparision allows the expected user experience.
+    if ($cursorPosition -gt $line.Length) {
+        $line = "$line "
+    }
+
+    # Mock bash with environment variable settings
+    New-Item -Path Env: -Name _ARGCOMPLETE -Value 1 | Out-Null # Enables tab complition in argcomplete
```

Comment:
This is a very creative approach using the bash completer, however, since these env vars are for the process and gets inherited by `az` and then removed, it would be better to use .NET API to start the process and leveraging https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.environmentvariables?view=net-6.0#system-diagnostics-processstartinfo-environmentvariables

It would look something like:

```powershell
$env = @{
  _ARGCOMPLETE = 1
  COMP_TYPE = 9
  _ARGCOMPLETE_IFS = " "
}

$psi = [System.Diagnostics.ProcessStartInfo]::new($AzCommand)
$psi.RedirectStandardError = $true
$psi.RedirectStandardOutput = $true

foreach ($var in $env.keys) {
  $psi.Environment.Add($var, $env.$var)
}

$az = [System.Diagnostics.Process]::Start($psi)
$az.Start()
$az.WaitForExit()
$completionResult = $az.StandardOutput.ReadToEnd()
```